### ISSUE TITLE
Do Not Disturb in macOS 12 Monterey

### ISSUE THREAD

**Issue by AurevoirXavier** [Jean-Tinland/simple-bar#249]
Several new modes

![image](https://user-images.githubusercontent.com/16152305/141767759-cf32799b-d9c3-42f8-a877-ec59c33328fa.png)

---

This won't work anymore
![image](https://user-images.githubusercontent.com/16152305/141767801-a97e5671-495a-40fe-9ef0-a26559bcbd05.png)

**Comment by Jean-Tinland** [comment]
Indeed, It is a really simple implementation of a DnD toggle. I'll look into something in order to improve this.

**Comment by azinsharaf** [comment]
Do Not Disturb bell doesn't work on Ventura 13.0.1. Nothing happens when toggling it.

**Comment by Kn4ughty** [comment]
A way to do this would be with apple shortcuts. 

We could write a shortcut to change the focus setting and then also export the current setting so the widget can update appropriately.
I'm not quite sure how we export the current state,
Ideas are 
- Write the current state to a file with OSASCRIPT or just a shell script, which we then read.
- or if it's possible to tell simple-bar messages in OSASCRIPT we use that.  

This toggles dnd successfully but doesn't communicate its state correctly so it all breaks. 
<img width="640" alt="Screenshot 2023-03-28 at 12 13 40" src="https://user-images.githubusercontent.com/77908661/228101255-e52a1bf2-abf2-40c3-b78a-051617e33759.png">

Would probably require a rewrite of the dnd widget, and if we wanted it to work on older macOS versions we would need separate scripts for it too.

**Comment by Kn4ughty** [comment]
When I ran `npm i macos-focus-mode`  to install the package so I could use it in the project, I got this error in my terminal.

<img width="927" alt="Screenshot 2023-04-07 at 19 38 45" src="https://user-images.githubusercontent.com/77908661/230585803-7e70db01-8433-4a3d-8c37-65babe3720c0.png">
The first error doesn't make sense to me unless it's because it's picky about how it's formatted.

The second error is that the config file package is just asking for a different package version than is installed with the project.
If I do `--force` then expectedly it breaks everything and the widget doesn't load anymore.
My npm is all up to date
I am new to the whole npm and javascript stuff, so if anyone could help me through it, I would appreciate it.
